corpus2ossfuzz/main.go

```diff
@@ -0,0 +1,106 @@
+// nolint: gosec
+
+// corpus2fuzz converts a corpus in native Go format to OSS-Fuzz format. It is
+// intended as a stop-gap until OSS-Fuzz is able to read native corpus files.
+//
+// See https://github.com/AdaLogics/go-fuzz-headers/blob/main/consumer.go for the
+// OSS-Fuzz data format.
+package main
+
+import (
+	"archive/zip"
+	"bytes"
+	"errors"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"path/filepath"
+	"strconv"
+)
+
+func main() {
+	corpusDir := flag.String("corpus", "", `the directory containing the Go corpus`)
+	output := flag.String("o", "", `the output zip file to write the OSS-Fuzz corpus`)
+	flag.Parse()
+
+	initCorpus(*corpusDir, *output)
+}
+
+func initCorpus(corpusDir, output string) {
+	log.SetFlags(0)
+	log.SetPrefix("corpus2ossfuzz: ")
+
+	entries, err := os.ReadDir(corpusDir)
+	if err != nil {
+		if os.IsNotExist(err) {
+			log.Printf("ignoring non-existing corpus directory %q", corpusDir)
+			os.Exit(0)
+		}
+		log.Fatal(err)
+	}
+
+	f, err := os.Create(output)
+	if err != nil {
+		log.Fatal(err)
+	}
+	z := zip.NewWriter(f)
```

Comment:
Missing `defer z.Close()`

Reply:
I don't believe it's necessary; a zip.Writer carries no resources other than memory.